archeryutils/handicaps/handicap_functions.py

```diff
@@ -12,754 +12,314 @@
-        score achieved on the round
-    rnd : rounds.Round
-        a rounds.Round class object that was shot
-    hc_sys : str
-        identifier for the handicap system
-    hc_dat : handicaps.handicap_equations.HcParams
-        dataclass containing parameters for handicap equations
-    arw_d : float or None, default=None
-        arrow diameter in [metres] default = None
-    int_prec : bool, default=False
-        display results as integers? default = False, with decimal to 2dp accuracy from
-        rootfinder
+    handicap_sys : Union[str, HandicapScheme]
+        String identifying the handicap scheme class to be generated, or an existing
+        HandicapScheme
+    \\**kwargs : float
+        Various kwargs defining custom values of parameters as appropriate for hc_sys
 
     Returns
     -------
-    handicap: int or float
-        Handicap. Has type int if int_prec is True, and otherwise has type false.
+    HandicapScheme
+        a subclass of HandicapScheme as appropriate for the inputs
 
     Raises
     ------
     ValueError
-        If an invalid score for the given round is provided.
+        If a handicap scheme is requested for which there is no defined class.
+
+    See Also
+    --------
+    handicap_scheme.HandicapScheme : The HandicapScheme class
+    handicap_scheme_agb.HandicapAGB :
+        The AGB HandicapScheme subclass and associated \\**kwargs
+    handicap_scheme_agb.HandicapAGBold :
+        The AGBold HandicapScheme subclass and associated \\**kwargs
+    handicap_scheme_aa.HandicapAA :
+        The AA HandicapScheme subclass and associated \\**kwargs
+    handicap_scheme_aa.HandicapAA2 :
+        The AA2 HandicapScheme subclass and associated \\**kwargs
 
     Examples
     --------
-    Angular deviation at a distance of 25m, using the AGB handicap system at a
-    handicap of 10 can be calculated with:
+    To generate a HandicapScheme class using the Archery GB (AGB) handicap scheme:
 
-    >>> import archeryutils as au
-    >>> hc_params = au.handicap_equations.HcParams()
-    >>> wa_outdoor = au.load_rounds.WA_outdoor
-    >>> au.handicap_functions.handicap_from_score(999, wa_outdoor.wa1440_90, "AGB", hc_params)
-    43.999964586102706
-    >>> au.handicap_functions.handicap_from_score(
-    ...     999,
-    ...     wa_outdoor.wa1440_90,
-    ...     "AGB", hc_params,
-    ...     int_prec=True
-    ... )
-    44.0
+    >>> from archeryutils import handicaps as hc
+    >>> agb_scheme = hc.handicap_scheme("AGB")
+    >>> agb_scheme
+    <archeryutils.handicaps.handicap_equations.HandicapAGB object at 0x31415ee00>
 
     """
-    # Check we have a valid score
-    max_score = rnd.max_score()
-    if score > max_score:
-        raise ValueError(
-            f"The score of {score} provided is greater than the maximum of {max_score} "
-            f"for a {rnd.name}."
-        )
-    if score <= 0.0:
+    if isinstance(handicap_sys, HandicapScheme):
+        return handicap_sys
```

Comment:
I really like this interface, returning the same scheme if you already happen to be holding on to one, but taking the strings as well. Very neat.